pkg/synchromanager/clustersynchro/cluster_monitor.go

```diff
@@ -61,26 +61,29 @@ func (synchro *ClusterSynchro) checkClusterHealthy() {
 	}
 
 	synchro.startResourceSynchro()
-	if lastReadyCondition.Status == metav1.ConditionTrue {
-		// TODO: if lastReadyCondition.Message != "", need process
-		return
-	}
-
+	message := "cluster health responded with ok"
 	condition := metav1.Condition{
 		Type:               clusterv1alpha2.ClusterHealthyCondition,
 		Status:             metav1.ConditionTrue,
 		Reason:             clusterv1alpha2.ClusterHealthyReason,
-		Message:            "cluster health responded with ok",
+		Message:            message,
 		LastTransitionTime: metav1.Now().Rfc3339Copy(),
 	}
-	defer func() {
-		synchro.healthyCondition.Store(condition)
-	}()
+
+	if lastReadyCondition.Status == metav1.ConditionTrue && lastReadyCondition.Message == message {
+		return
+	}
 
 	if _, err := synchro.dynamicDiscoveryManager.GetAndFetchServerVersion(); err != nil {
-		condition.Message = fmt.Sprintf("cluster health responded with ok, but get server version: %v", err)
-		klog.ErrorS(err, "Failed to get cluster version", "cluster", synchro.name)
+		message = fmt.Sprintf("cluster health responded with ok, but get server version: %v", err)
```

Comment:
message and condition.Message are not the same string.

You can create the metav1.Condition before setting it to `synchro.healthyCondition`

Reply:
yes~ fixed it ! Thank you